### PR TITLE
feat(PDS-53): Menu Component

### PR DESCRIPTION
**Pull request by Giogia**
### Description

Menu components implementation

### Issue ticket number and link

[Menù: Creazione Componente](https://makeitapp.atlassian.net/browse/PDS-53)

### TODO Checklist

- [ ] Make the collapse animation smoother

### CODE REVIEW

File: src/components/Menu/Menu.module.css
Comment (RiccardoCorona-MiaPlatform) [inline src/components/Menu/Menu.module.css:1]: Serve il file vuoto?

File: src/components/Menu/Menu.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import { Menu as AntMenu, MenuProps as AntMenuProps } from 'antd'
+import { ReactElement, useMemo } from 'react'
+import classnames from 'classnames'
+
+import capitalizeGroupLabels from './Menu.utils'
+import styles from './Menu.module.css'
+
+const { menu } = styles
```
Comment (RiccardoCorona-MiaPlatform) [inline src/components/Menu/Menu.tsx:8]: Da rimuovere?

File: package.json [hunk 2]
```diff
@@ -69,6 +69,8 @@
     "@typescript-eslint/eslint-plugin": "^6.0.0",
     "@typescript-eslint/parser": "^6.0.0",
     "@vitejs/plugin-react-swc": "^3.3.2",
+    "autoprefixer": "^10.4.15",
```
Comment (Giogia) [inline package.json:72]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -80,6 +82,7 @@
     "jest-environment-jsdom": "^29.6.2",
     "jest-transformer-svg": "^2.0.1",
     "jest-watch-typeahead": "^2.2.2",
+    "postcss": "^8.4.29",
```
Comment (Giogia) [inline package.json:85]:
```suggestion
```

File: src/components/Menu/Menu.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+import { category, divider, group, item } from './Menu.mocks'
+import { Menu } from './'
+
+const figmaUrl = 'https://www.figma.com/file/0ar6alIEDe8iYCb8kU7Rxd/%5BDS%5D-Console-Mia-Platform?node-id=987%3A10396&mode=dev'
+
+const meta = {
+  component: Menu,
+  parameters: {
+    design: { type: 'figma', url: figmaUrl },
+  },
```
Comment (Giogia) [inline src/components/Menu/Menu.stories.tsx:32]:
```suggestion
const meta = {
  component: Menu,
```

Comment (Giogia) [reply src/components/Menu/Menu.stories.tsx:32]: Let's remove the figma URL for the moment

File: src/components/Menu/Menu.test.tsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+import { render, waitFor } from '@testing-library/react'
+
+import { category, divider, group, item } from './Menu.mocks'
+import { Menu } from '.'
+import { Modes } from './Menu.types'
+
+const { Vertical } = Modes
+
+describe('Menu Component', () => {
+  const originalErr = global.console.error.bind(global.console.error)
```
Comment (Giogia) [inline src/components/Menu/Menu.test.tsx:28]: Could you please add a comment to explain the reason behind this warning suppression?

File: src/components/Menu/Menu.test.tsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+  afterAll(() => {
+    global.console.error = originalErr
+  })
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+  const items = [item, group, divider, category]
```
Comment (Giogia) [inline src/components/Menu/Menu.test.tsx:43]: What do you think about moving this before the describe?

File: src/components/Menu/Menu.theme.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * Generates a Ant theme configuration for Menu component based on a theme configuration (default theme).
+ *
+ * @link https://ant.design/components/menu#design-token
+ *
+ * @param {Partial<Theme>} theme - theme configuration.
+ * @returns {Partial<ComponentsTheme>} The generated Menu Ant theme configuration.
+ */
+export default ({ palette, shape, spacing, typography }: Partial<Theme>): ComponentsTheme['Menu'] => ({
+  groupTitleColor: palette?.common?.grey[700],
```
Comment (Giogia) [inline src/components/Menu/Menu.theme.ts:31]:
```suggestion
  groupTitleColor: palette?.common?.grey?.[700],
```

File: src/components/Menu/Menu.theme.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  groupTitleFontSize: parse(typography?.caption?.fontSize),
+  groupTitleLineHeight: 1,
+  itemBg: 'transparent',
+  itemColor: palette?.common?.grey[800],
+  itemHeight: (parse(typography?.bodyS?.lineHeight) ?? 24) + (parse(spacing?.padding?.sm) ?? 8),
+  itemHoverBg: '#1890FF15',
+  itemHoverColor: palette?.common?.grey[800],
+  itemSelectedBg: palette?.common?.white,
+  itemSelectedColor: palette?.common?.grey[800],
```
Comment (Giogia) [inline src/components/Menu/Menu.theme.ts:57]:
```suggestion
  groupTitleColor: palette?.common?.grey?.[700],
  groupTitleFontSize: parse(typography?.caption?.fontSize),
  groupTitleLineHeight: 1,
  itemBg: 'transparent',
  itemColor: palette?.common?.grey?.[800],
  itemHeight: (parse(typography?.bodyS?.lineHeight) ?? 24) + (parse(spacing?.padding?.sm) ?? 8),
  itemHoverBg: '#1890FF15',
  itemHoverColor: palette?.common?.grey?.[800],
  itemSelectedBg: palette?.common?.white,
  itemSelectedColor: palette?.common?.grey?.[800],
```

File: src/components/Menu/Menu.tsx [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+  return (
+    <ConfigProvider theme={{ components: { Menu: menuTheme } }}>
+      <Skeleton
+        active
+        loading={isLoading}
+        paragraph={{ rows: 6, width: ['30%', '80%', '65%', '30%', '70%', '60%'] }}
+      >
+        <AntMenu
+          className={classnames([menu])}
```
Comment (Giogia) [inline src/components/Menu/Menu.tsx:133]:
```suggestion
          className={menu}
```

Maybe we can save some computation this way

File: src/components/Menu/Menu.types.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MenuItemType } from 'antd/es/menu/hooks/useItems'
+
+export type Item = { type?: string, children?: Item[] } & MenuItemType
```
Comment (Giogia) [inline src/components/Menu/Menu.types.ts:21]:
```suggestion
export type Item = MenuItemType & { 
   type?: string, 
   children?: Item[] 
}
```

Comment (Giogia) [reply src/components/Menu/Menu.types.ts:21]: I think this way is more clear, not necessary though

File: src/components/Menu/Menu.utils.tsx [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+ * @param {items} items - Menu items to format.
+ * @param {selectedItem} selectedItem - The currently selected menu item.
+ * @param {isCollapsed} isCollapsed - Whether the menu is collapsed.
+ *
+ * @returns {ItemType[]} array of formatted menu items.
+ */
+function formatLabels(
+  items: Item[] = [],
+  selectedItem: string | undefined,
+  isCollapsed: boolean | undefined
```
Comment (Giogia) [inline src/components/Menu/Menu.utils.tsx:35]:
```suggestion
  selectedItem?: string,
  isCollapsed?: boolean
```

File: vite.config.ts [hunk 12]
```diff
@@ -1,3 +1,4 @@
+import autoprefixer from 'autoprefixer'
```
Comment (Giogia) [inline vite.config.ts:1]: since we are using CSS modules, this is not needed anymore

File: src/components/Menu/Menu.theme.ts (hunk 8 above)
Comment (Giogia) [reply src/components/Menu/Menu.theme.ts:57]: I added the safe access ?.['number'] to properties, then I removed the common items (primaryTheme inherits from the default theme) and added transparent background to sub menus

**Comment by Giogia** [comment]
lgtm

**Review by RiccardoCorona-MiaPlatform — APPROVED** [review]

**Pull request merged**